### DISCUSSION TITLE
FreeSewing org and dev website organization, layout, content

### DISCUSSION THREAD

**Discussion by BenJamesBen** [freesewing/freesewing#3988] [Forum]
Below is a possible brainstorm for organizing FreeSewing.org documentation and website. It's a bit jumbled because it mixes content and navigation, but overall I think it makes some sense. 

The main points is that most documentation is pulled out of the Documentation section and is moved to the sections they are related to. For example, design instructions/cutlists/etc. are moved to the individual designs. Measurement documentation is moved to the measurements section, which is in the How to use this Site section. The Pattern Notation Guide is moved into the section that discusses patterns. The Revenue Pledge is moved into the About FreeSewing section. Etc.

The only items left in Documentation are Frequently Asked Questions and Sewing, and so the section is renamed "Other Documentation".

---

FreeSewing.org home page:
  - Succinct info about what FreeSewing is and what it can do for you.
  - (A few, random Designs. "some of our designs".)
  - (A few, recent Blog and Showcase entries, with links to Blog and Showcase)
  - (excerpt from the most recent newsletter, with links to Newsletter archive)
  - For makers, For designers, For developers
  - Join us on Discord.
  - Subscribe to our newsletter
  - Support FreeSewing.

Main/Top-level Navigation:
  - How to use FreeSewing
  - Our Designs (Patterns)
  - About Freesewing
  - Other Documentation
  - Blog
  - Getting Help
  - Account

---

How to use FreeSewing:
  - Webpage:
    - Quickstart (step by step instructions)
      1. Sign up for an account
      2. Enter your measurements
      3. Generate a pattern
      4. Optional adjustments
      5. Export to PDF
  - Navigation:
  - Quickstart (same as above)
  - All about accounts
    - (Documentation for accounts and the account website UI goes here)
    - (How to manage saved patterns and measurement sets.)
  - Measurement Sets
    - (Documentation about measurements go here.)
    - (Possibly also documentation about the standard sizes.)
    - Sizing tables
  - Patterns
    - (Documentation about the lab UI (or the equivalent org version) goes here.)
    - Pattern notation guide goes here.

---

Our Designs (Patterns):
  - Webpage:
    - top of the page explains why we use "Designs" instead of "Patterns"
    - (a few random showcase images)
    - all our Designs
    - Showcase
  - Navigation:
    - Designs
      - (individual Designs)
    - Showcase

Example Individual Design webpage:
  - Name and description of the design, with diagrams and info (same as existing website)
  - (The next 6 or so items are Documentation. The order is customizable. Additional doc pages can be included by the designer ("Tutorials", "Historic Background", "Links to videos", etc.)
  - What you need
  - Required Measurements
  - Pattern options
  - Fabric options
  - Cutting instructions
  - Construction instructions
  - Showcase photos or example photos

---

Other Documentation:
  - Webpage:
    - Links to top 10 docs: (what is FreeSewing, What is FreeSewing.org, etc.)
    - Please also see "How to Use FreeSewing"
    - Please also see "Measurements" (which is in the How to Use FreeSewing section, not in Documentation)
    - Please also see "Patterns" (which is in the How to Use FreeSewing section, not in Documentation)
    - Please also see individual designs, because the documentation for designs is with each design, not here.
  - Navigation:
    - Frequently Asked Questions
      - How do I create my own Designs? (Developer documentation moved to the FAQ section)
      - cisMale/cisFemale wording.
      - What are made-to-measure sewing patterns?
    - Sewing

---

About FreeSewing
  - About Us
    - What is FreeSewing
    - What is FreeSewing.org
    - Who we are (contributors and patrons)
    - Community standards
    - Revenue pledge
    - Privacy notice
    - Licensing info (open source info)
    - Your rights
    - Release notes (current release info)
    - Thank you to our sponsors (Vercel, Algolia, Crowdin, Bugsnag? etc.)
  - How to Participate
    - Discord
    - Contributor Calls
    - Social Media and hashtags (where to find us)
    - Newsletter
  - How to Help
    - Ways to contribute (moved or split from the dev docs)
    - Become a patron
    - Share FreeSewing
  - Our Teams
    - Community (newsletter, blog, showcase, social media, outreach)
    - Pattern Design
    - Development
    - Documentation
    - Translation

---

Account:
  - Your Measurement Sets
  - Your Patterns
  - Profile and Settings
  - Account Actions

**Comment by BenJamesBen** [comment]
From #3386. Please also see that Issue for additional content details, intended audience, and theory about the content that the dev website should contain.

A proposed organizational structure for the FreeSewing.dev website. This structure more naturally accommodates the documentation additions I would like to add.

FreeSewing.dev:
- For all Developers
- For Pattern Designers and Coders
- For Infrastructure Coders
- For Writers
- For Translators
- How to work as a team and as part of a greater community
- About FreeSewing
- Development Blog

---

- For all Developers
  - Lab UI documentation
  - Getting Started
    - Linux tutorial
    - Mac tutorial
    - Windows tutorial
    - Gitpod Tutorial
    - Codespaces Tutorial
    - Vercel Tutorial
  - Using the development environment
    - yarn kickstart
    - yarn lab to start the lab dev environment
    - cd sites/dev ; yarn start to start the dev website dev environment
    - cd sites/org ; yarn start to start the org website dev environment
    - Using web browsers (clear cache, cookies, local storage; using Javascript console)
    - yarn clean
    - npm rebuild sharp
    - ? Node and nvm? How to change node versions?
  - The `freesewing/freesewing` repository.
    - Map/tree showing overview of where files are located
    - What files get generated automatically, and what files do you edit instead?
  - Basic workflows
    - Keeping your repository up-to-date
    - Dealing with git pull conflicts
    - Using branches to organize changes
    - Running lint
    - Testing changes with `yarn test`
  - Git
    - Basic concepts: repositories, commits, branches, etc.
    - Quick reference
    - Common git challenges
      - Adding files
      - Moving files
      - Deleting files
      - Merging conflicts
      - Diffing files
      - Viewing history
    - Git Training
    - Useful advanced tricks (aliases, config, replacing commit messages)
  - GitHub
    - Account basics
      - How to set up and use those security login tokens.
    - Working with PRs
      - How to submit PR
      - Dealing with with PR problems
        - Test failures
        - Lint failures
        - Deepscan failures
        - Merge conflictss
      - How to review a PR (including viewing Vercel deployment, if available)
      - How to make requested changes to a PR and resubmit
      - Working with Draft PRs
      - GitHub labels and the PR state workflow.
    - Working with Issues/Bugs
      - How to file a bug (including best practices)
      - How to work on a bug
    - Working with Discussions
    - GitHub Tutorial
    - FreeSewing's GitHub configuration and integrations. The other freesewing repositories.
    - Finding what you want. (How to search, how to filter views, , using the Project pages, etc.)
  - ?SVG?
    - ?SVG basics?
  - Other useful information
    - How to add new Contributors.

---

- For Pattern Designers and Coders
  - Before you start (SVG, coordinates, units, Béziers)
  - Pattern design tutorial
  - Design submission process (how to submit new designs to FreeSewing)
  - Pattern concepts guide (sets, stacks, parts, points, etc.)
  - Reference
    - Core API
    - Macros (includes Plugins)
    - Snippets (includes Plugins)
    - Other Plugins (svgattr, theme, versionfree-svg, timing, i18n)
    - Measurements
    - Settings
    - CSS classes
  - Common Pattern Design and Coding challenges and howtos
    - How to actually add translation text, after including translation keys in the pattern.
  - Common Pattern Documentation challenges and howtos
    - How to write documentation for new designs.
    - How to add the initial translation files for new designs.
    - How to generate those options images.
  - Pattern design best practices
  - Pattern design Frequently Asked Questions
  - Best practices for design code reviews, howto

---

- For Advanced or Infrastructure Coders
  - Overview of how FreeSewing works (backend, posts)
  - Technologies used (Algolia, OpenGraph etc.)
    - Node.js, Next.js, React, MDX, Strapi, Husky, Prettier, Remark, Rehype, Sanity, Terraform, Eslint, Ansible, Lerna
    - Nx
    - Concepts. (workbench, lightbox, worm, slug)
  - The various *.freesewing.* sites.
  - FreeSewing Backend API
  - Lifecycle hooks
  - Freesewing's packages
    - Package documentation
  - Plugin guide
  - ? Deploying FreeSewing as a node app?
  - ? Generating patterns with a browser?
  - Writing tests. Playwright. Chai Expect.
  - FreeSewing's release process and workflow
    - How do the org and dev sites get updated?
    - How do you find the current version in GitHub?
  - The package.json commands and /scripts. What do they do?
  - How does the newsletter get published/emailed?
  - Infrequently Asked Questions
    - What is that `exceptions.yaml` file, and how is it used?
    - How is prettier configured?
    - How is eslint configured?

---

- For Writers
  - CSS classes
  - Common Editing challenges
  - Markdown Guide
  - Style Guide
  - How to make SVG illustrations for the website.
  - Strapi Guide/tutorial
  - Sanity Guide/tutorial
  - How to submit an edit to a documentation page (for contributors who want to help change documentation but who don't want to learn git or GitHub)
  - Best practices for documentation reviews
  - How to add a showcase?
  - A brief summary of the translation workflow that occurs after a documention change/addition is made. This is so people are aware of what happens.

---

- For Translators
  - ? The translation process/workflow. A more detailed description for the people actually working on translations.
    - Including the Crowdin PRs process
  - ? Crowdin
  - ? Strapi
  - ? Translation Guide
  - ? i18n strings: location of files, how to edit

---

- How to work as a team and as part of a larger community
  - Communication
    - Modes: Discord. GitHub reviewers. Tagging people on GitHub.
    - Make sure to communicate: after filing a bug, before closing a bug, after filing a PR.
  - Collaboration
    - Using Discord voice chat.
      - Sharing screens.
  - Consensus
  - Stakeholders
    - Identifying who the stakeholders are.
  - No middle managers

---

- About FreeSewing
  - How to contribute (the developer-facing ways) (Or, should this just be on org?)
  - Code of conduct (actually, this should be on the org site. Or both sites?)
  - Community standards (from org. another item that should be on both sites?)
  - Licenses, copyright, and Open Source notices (another on both site?)
  - Sponsors (another on both sites?)

**Comment by BenJamesBen** [comment]
From #3331: 

Add Howto for how to communicate information to users

Intended audience is pattern designers/coders, for how to provide info to end users. For example, how to tell them how much elastic to use. How to tell them the dimensions of critical parts (aside from using the usual `paperless` dimensions). How to provide warning messages and tips. Etc.

- How to use the `log` methods.
- How to print text directly to the generated pattern, with a bounding box.
- How to use `i18n` in pattern text, log messages, etc. (An actual working example, from start to end, with all the code and all the translation files. Not just telling the developer to "use translation keys".)

**Comment by BenJamesBen** [comment]
From #3392. 

Establish a place for UI documentation

In the December 10, 2022 Contributor Call, we determined that we would establish place in the documentation for UI docs. This was because I was going to put them in the Reference API section under Part Options, but we determined that this section would be strictly for the FreeSewing core. (This was in response to PR https://github.com/freesewing/freesewing/pull/3169 .)

Please see #3392 for discussion about instead putting this info into the Core API section (which is where it logically belongs, even though it is not strictly part of Core).

**Comment by BenJamesBen** [comment]
From #3715:

Designs documentation should include information about how how options actually affect the measurements of the garments. Not just larger or smaller, but exactly how much larger or how much smaller.

**Comment by BenJamesBen** [comment]
Customers using Ubuntu or Debian can run into an issue where they find that a different `yarn` command has been added via the `cmdtest` package. This should be documented in the Getting Started tutorial under a `<Note>` or `<Warning>`.

https://discord.com/channels/698854858052075530/698862765053575188/1114610448613003416
discord://discord.com/channels/698854858052075530/698862765053575188/1114610448613003416
https://github.com/yarnpkg/yarn/issues/3189

Also, until https://github.com/freesewing/freesewing/issues/3979 is fixed, we should add a howto or somesuch about how to rebuild `sharp`. Basically, we need to implement https://github.com/freesewing/freesewing/issues/3979.

**Comment by BenJamesBen** [comment]
- If you rename a directory and the directory's tree containing translated files, it triggers a re-translation in Crowdin for all those translation files. Beware that this could cause us to go over quota.

**Comment by joostdecock** [comment]
I suspect this is superseded by the more recent umbrella issues for FreeSewing.org
